sdk/core/src/error/mod.rs

```diff
@@ -0,0 +1,371 @@
+use std::borrow::Cow;
+use std::fmt::{Debug, Display};
+
+mod azure_core_errors;
+mod hyperium_http;
+mod macros;
+
+/// A convience alias for `Result` where the error type is hard coded to `Error`
+pub type Result<T> = std::result::Result<T, Error>;
+
+/// The kind of error
+///
+/// The classification of error is intentionally fairly coarse.
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum ErrorKind {
+    /// An HTTP status code that was not expected
+    HttpResponse { status: u16 },
+    /// An error performing IO
+    Io,
+    /// An error converting data
+    DataConversion,
+    /// An error getting an API credential token
+    Credential,
+    #[cfg(feature = "mock_transport_framework")]
+    /// An error having to do with the mock framework
+    MockFramework,
+    /// A catch all for other kinds of errors
+    Other,
+}
+
+impl ErrorKind {
+    pub fn http_response(status: u16) -> Self {
```

Comment:
Related above, both `status` and `code` (or `error_code`) are required by Azure standard error responses and should be, at a minimum, both captured.